eth/backend.go

```diff
@@ -620,6 +623,29 @@ func (s *Ethereum) NetPeerCount() (uint64, error) {
 	return sentryPc, nil
 }
 
+func (s *Ethereum) NodesInfo(limit int) (*remote.NodesInfoReply, error) {
+	if limit == 0 || limit > len(s.sentries) {
+		limit = len(s.sentries)
+	}
+
+	nodes := make([]*prototypes.NodeInfoReply, 0, limit)
+	for i := 0; i < limit; i++ {
+		sc := s.sentries[i]
+
+		node, err := sc.NodeInfo(context.Background(), nil)
+		if err != nil {
+			log.Error("sentry nodeInfo", "err", err)
+		}
+
+		nodes = append(nodes, node)
+	}
+
+	nodesInfo := &remote.NodesInfoReply{NodesInfo: nodes}
+	sort.Sort(nodesInfo)
```

Comment:
No deduplication after sorting? :) If we have multiple sentries, we may see a node multiple times

Reply:
Sorry, I didn't expect that, maybe i just do not fully understand the whole system. Is it enough to check `name` & `enode` for deduplication?

Reply:
Since `enode` should be uniquely identifying a node, it should be enough to deduplicate by it. But I could not find quickly, what does this sorting sort by? Because if it sorts by enode first, then we could deduplicate simply by comparing adjacent records

Reply:
I am happy to do dedup in separate PR though, once we have observed the duplicates

Reply:
it sorts by [(name, enode)](https://github.com/ledgerwatch/erigon-lib/blob/main/gointerfaces/remote/sort.go#L12) as requested [here](https://github.com/ledgerwatch/erigon/issues/2679). I'll do dedup by these two fields, np.

Reply:
just wondering if two nodes can have different names, but the same enode? And if they do, should they be considered the same?

Reply:
theoretically they may appear with different names to different sentries, while having enode. We can dedup by two fields, this should work

Reply:
Done.